src/main/java/org/logstash/tcp/InputLoop.java

```diff
@@ -0,0 +1,125 @@
+package org.logstash.tcp;
+
+import io.netty.bootstrap.ServerBootstrap;
+import io.netty.buffer.ByteBuf;
+import io.netty.channel.ChannelFuture;
+import io.netty.channel.ChannelHandler;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelInboundHandlerAdapter;
+import io.netty.channel.ChannelInitializer;
+import io.netty.channel.ChannelOption;
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.nio.NioEventLoopGroup;
+import io.netty.channel.socket.SocketChannel;
+import io.netty.channel.socket.nio.NioServerSocketChannel;
+import java.io.Closeable;
+
+/**
+ * Plain TCP Server Implementation.
+ */
+public final class InputLoop implements Runnable, Closeable {
+
+    /**
+     * Netty Boss Group.
+     */
+    private final EventLoopGroup boss;
+
+    /**
+     * Netty Worker Group.
+     */
+    private final EventLoopGroup worker;
+
+    /**
+     * The Server Socket's {@link ChannelFuture}.
+     */
+    private final ChannelFuture future;
+
+    /**
+     * Ctor.
+     * @param host Host to bind the listen to
+     * @param port Port to listen on
+     * @param decoder {@link Decoder} provided by Jruby
+     */
+    public InputLoop(final String host, final int port, final Decoder decoder) {
+        worker = new NioEventLoopGroup();
```

Comment:
@jordansissel said so right (Jordan please confirm :D)?

I have no idea either, my gut feeling would be `1` to avoid negatively affecting the overall system under pressure, but that's way too slow here :(
Maybe this value is ok, given that the main Queue blocks anyway and we will balance things out this way?